test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ValidateTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Rest;
+using Microsoft.Health.Fhir.Tests.Common;
+using Microsoft.Health.Fhir.Tests.Common.FixtureParameters;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Rest
+{
+    [Trait(Traits.Category, Categories.Validate)]
+    [HttpIntegrationFixtureArgumentSets(DataStore.All, Format.All)]
+    public class ValidateTests : IClassFixture<HttpIntegrationTestFixture>
+    {
+        private readonly HttpClient _client;
+        private readonly JsonSerializer _serializer;
+
+        public ValidateTests(HttpIntegrationTestFixture fixture)
+        {
+            _client = fixture.HttpClient;
+            _serializer = JsonSerializer.Create();
+        }
+
+        [Theory]
+        [InlineData("Patient/$validate", "{\"resourceType\":\"Patient\",\"name\":{\"family\":\"test\",\"given\":\"one\"}}")]
+        [InlineData("Observation/$validate", "{\"resourceType\":\"Observation\",\"status\":\"registered\",\"code\":{\"coding\":[{\"system\":\"system\",\"code\":\"code\"}]}}")]
+        public async void GivenAValidateRequest_WhenTheResourceIsValid_ThenAnOkMessageIsReturned(string path, string payload)
+        {
+            HttpResponseMessage response = await _client.SendAsync(GenerateValidateMessage(path, payload));
+
+            var contentString = await response.Content.ReadAsStringAsync();
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            CheckOperationOutcomeIssue(
+                contentString,
+                OperationOutcome.IssueSeverity.Information,
+                OperationOutcome.IssueType.Informational,
+                "All OK");
+        }
+
+        [Theory]
+        [InlineData(
+            "Patient/$validate",
+            "{\"resourceType\":\"Patient\",\"name\":\"test, one\"}",
+            "Type checking the data: Since type HumanName is not a primitive, it cannot have a value (at Resource.name[0])")]
+        [InlineData(
+            "Observation/$validate",
+            "{\"resourceType\":\"Observation\",\"code\":{\"coding\":[{\"system\":\"system\",\"code\":\"code\"}]}}",
+            "Element with min. cardinality 1 cannot be null")]
```

Comment:
which element?

Reply:
The name/path is in another parameter of the returned operation outcome. I added another test to cover this case.